lib/checker.js

```diff
@@ -60,15 +93,22 @@ Checker.prototype.checkDirectory = function(path) {
             }
 
             return vowFs.stat(fullname).then(function(stat) {
+                var promise;
+
                 if (stat.isDirectory()) {
                     return this.checkDirectory(fullname);
                 }
 
                 if (!this._hasCorrectExtension(fullname)) {
-                    return [];
+                    if (this._configuration.getExtract() === 'never') {
+                        return [];
+                    }
+                    promise = Vow.when(this.extractFile(fullname));
```

Comment:
Wouldn't you need to check that `fullname` contains the `.html` extension? Is that check done somewhere else? Seems like this would attempt to parse *everything* but `.js` files.